mmf/utils/build.py

```diff
@@ -165,6 +186,16 @@ def build_dataloader_and_sampler(
     if not isinstance(dataset_instance, torch.utils.data.IterableDataset):
         other_args = _add_extra_args_for_dataloader(dataset_instance, other_args)
 
+    if is_xla():
+        shuffle = True
```

Comment:
This needs to be always set to True?

Reply:
It's good callout, thank you. I think we should use others["shuffle'] instead of creating a local variable and forcing to True always. I just made this change, pushing shortly.

Reply:
done.